src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -401,25 +425,66 @@ async def package_python_dist(
         ),
     )
 
+    # Find the source roots for the build-time 1stparty deps (e.g., deps of setup.py).
```

Comment:
This is a pretty big rule, consider factoring out a `@rule_helper`. (Not sure that's the right call)

Reply:
This whole file is huge and needs refactoring, but I will hold off on that for now.